Correctly define dependencies for pytest and plugins

gsnedders opened the pull request:
#### 1819e751cfdc66fb2dccb643efdabc7d4b7850b1
<pre>
Correctly define dependencies for pytest and plugins
<a href="https://bugs.webkit.org/show_bug.cgi?id=253642">https://bugs.webkit.org/show_bug.cgi?id=253642</a>

Reviewed by Fujii Hironori.

* Tools/Scripts/webkitpy/__init__.py:

Canonical link: <a href="https://commits.webkit.org/261467@main">https://commits.webkit.org/261467@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8ad21fe48cfb4cb0f2b7fa2d4fe105c26267eb93

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/111692 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/20828 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/306 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/3504 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120425 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/115753 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/22184 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/11908 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/86/builds/3239 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117458 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/16471 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99618 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/104657 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/115212 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/98423 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/197 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/45418 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/13306 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/195 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/92289 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13796 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/9641 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/19242 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52192 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7977 "'git push ...'") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15787 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Tools/Scripts/webkitpy/__init__.py:
```diff
@@ -36,9 +36,15 @@
+        Package("pytest", Version(7, 2, 0),
+                implicit_deps=["attr", "pluggy", "iniconfig"]
+                + (["exceptiongroup"] if sys.version_info < (3, 11) else [])
+                + (["importlib_metadata"] if sys.version_info < (3, 8) else [])
```
I'm not a Python expert. Do we need to change the following line to add `<3.8` to align with this?
https://github.com/WebKit/WebKit/blob/3cc2ae9e348aa7ff1ff4487f812d8ba2ba2022c4/Tools/Scripts/webkitpy/__init__.py#L53-L54
~~~python
if sys.version_info >= (3, 6):
    if sys.version_info < (3, 8):
       AutoInstall.register(Package('importlib_metadata', Version(4, 8, 1)))
~~~

gsnedders replied:
I just made this match what pytest requires in its setup.cfg; it might mean we're unnecessarily declaring `importlib_metadata` on 3.8+, I haven't looked to see if there's other usage?

gsnedders committed:
Correctly define dependencies for pytest and plugins
https://bugs.webkit.org/show_bug.cgi?id=253642

Reviewed by Fujii Hironori.

* Tools/Scripts/webkitpy/__init__.py:

Canonical link: https://commits.webkit.org/261467@main
